sponsorship_compassion/tests/test_sponsorship_compassion.py

```diff
@@ -0,0 +1,459 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Albert SHENOUDA <albert.shenouda@efrei.net>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp.tests import common
+from datetime import datetime
+from openerp import netsvc
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+import logging
+logger = logging.getLogger(__name__)
+
+
+class test_sponsorship_compassion(common.TransactionCase):
+
+    def setUp(self):
+        super(test_sponsorship_compassion, self).setUp()
+        # Retrieve receivable and payable accounts
+        account_type = self.registry('account.account.type').search(
+            self.cr, self.uid, [
+                ('code', '=', 'receivable'),
+            ])[0]
+        property_account_receivable = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'receivable'),
+                ('user_type', '=', account_type)
+            ])[0]
+        account_type = self.registry('account.account.type').search(
+            self.cr, self.uid, [
+                ('code', '=', 'payable'),
+            ])[0]
+        property_account_payable = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'payable'),
+                ('user_type', '=', account_type)])[0]
+        property_account_income = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'other'),
+                ('name', '=', 'Property Account Income Test')
+            ])[0]
+        property_account_expense = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'other'),
+                ('name', '=', 'Property Account Expense Test')
+            ])[0]
+        # Creation of partners
+        partner_obj = self.registry('res.partner')
+        self.partner_id = partner_obj.create(self.cr, self.uid, {
+            'name': 'Monsieur Pumba',
+            'property_account_receivable': property_account_receivable,
+            'property_account_payable': property_account_payable,
+            'notification_email_send': 'comment',
+        })
+        # Retrieve a payment term
+        payment_term_obj = self.registry('account.payment.term')
+        self.payment_term_id = payment_term_obj.search(self.cr, self.uid, [
+            ('name', '=', '15 Days')
+        ])[0]
+        product_obj = self.registry('product.product')
+        product_obj.write(self.cr, self.uid, 90, {

```

Comment:
You should search for the product you are looking for instead on writing on a given ID, as you are not guaranteed that the product will have the same ID on all test databases...